### PR TITLE
core/server: sharing failing test

### PR DESCRIPTION
**Pull request by tvanderstad**

### CODE REVIEW

File: core/libs/shared/src/core_ops.rs [hunk 1]
```diff
@@ -71,9 +71,10 @@ where
             return Err(SharedError::InsufficientPermission);
         }
 
+        let parent_owner = self.find(parent)?.owner().0;
         let parent_key = self.decrypt_key(parent, account)?;
-        let new_file =
-            FileMetadata::create(pub_key, *parent, &parent_key, name, file_type)?.sign(account)?;
+        let new_file = FileMetadata::create(&parent_owner, *parent, &parent_key, name, file_type)?
+            .sign(account)?;
```
Comment (tvanderstad) [inline core/libs/shared/src/core_ops.rs:77]: Looks good, prob need something similar for move

File: server/server/src/error_handler.rs [hunk 2]
```diff
@@ -170,6 +170,7 @@ impl From<SharedError> for ServerError<DeleteAccountError> {
 
 impl From<SharedError> for ServerError<UpsertError> {
     fn from(err: SharedError) -> Self {
+        // panic!("{err}");
```
Comment (tvanderstad) [inline server/server/src/error_handler.rs:173]: note

File: core/libs/shared/src/server_tree.rs [hunk 3]
```diff
@@ -13,11 +14,40 @@ where
...
     pub owned: &'a mut TransactionTable<'b, Owner, HashSet<Uuid>, Log1>,
     pub metas: &'a mut TransactionTable<'b, Uuid, ServerFile, Log2>,
 }
 
+impl<'a, 'b, Log1, Log2> ServerTree<'a, 'b, Log1, Log2>
+where
+    Log1: SchemaEvent<Owner, HashSet<Uuid>>,
+    Log2: SchemaEvent<Uuid, ServerFile>,
+{
+    // todo: optimize/cache
```
Comment (tvanderstad) [inline core/libs/shared/src/server_tree.rs:27]: @Parth 🙏

Comment (tvanderstad) [reply core/libs/shared/src/server_tree.rs:27]: We need a cache that maps users to ids of files shared with them

**Review by Parth — APPROVED** [review]

**Pull request merged**